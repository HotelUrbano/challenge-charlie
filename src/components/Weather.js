import React from 'react'

//esse componente vai acabar sendo stateless pelo visto, pode ser reescrito para melhor entendimento
class Weather extends React.Component{
    state = {
        scale : "C",
    }
    render(){
        return (
            <div>
                <p>{this.props.day}</p>
            {
                //https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
                this.props.temperature && <p>Temperatura: {this.showTemperature()} Â°{this.state.scale} </p> 
            }
            {
                this.props.weather && <p>Clima: {this.props.weather} </p> 
            }
            {
                this.props.wind && <p>Vento: {this.props.wind} km/h</p> 
            }
            {
                this.props.humidity && <p>Humidade: {this.props.humidity} %</p> 
            }
            {
                this.props.pression && <p>Pressao: {this.props.pression} mb</p> 
            }
  
            </div>
        );
    }
    //mostra a temperatura em celsius oum fahrenheit
    showTemperature(){
        if(this.state.scale === "C"){
            return this.props.temperature;
        } else {
            let temperatureImperial = this.props.temperature;
            temperatureImperial = parseFloat(temperatureImperial);
            temperatureImperial = (temperatureImperial / 5) * 9 + 32;
            return temperatureImperial;
        }
    }
}
export default Weather