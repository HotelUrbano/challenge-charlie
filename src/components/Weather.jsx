import React from 'react'
import 'modules/bootstrap/dist/css/bootstrap.min.css'
import 'modules/font-awesome/css/font-awesome.min.css'
import Condition from '../pages/Condition'


export default props => (
    <div className=''>
        <Condition />
    </div>
)
