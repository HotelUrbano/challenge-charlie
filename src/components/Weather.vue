<template>
    <main class="content">
        <h1 class="colors"><span class="icon-font">( </span>{{weather.city}}, {{weather.region}}</h1>
        <div class="colors todayC">
            <span class="icon-font">
                {{weather.condition | iconWeather}}
            </span>

            <span>{{weather.todayC}}</span><span>{{weather.today}}</span>
            {{weather.condition | translate}}
            Vento:{{weather.windDirection}} {{weather.windSpeed}}
            Humidade:{{weather.humidity}}
            Pressão: {{weather.pressure}}
        </div>
        <div class="colors tomorrowC">
            Amanhã
            <span>{{weather.tomorrowC}}</span>
            <span>{{weather.tomorrow}}</span>
        </div>

        <div class="colors afterC">
            Depois de amanhã
            <span>{{weather.afterC}}</span>
            <span>{{weather.after}}</span>
        </div>
    </main>
</template>
<script>

export default {
    name: "Weather",
    props: ["weather", "lat", "long"],
    updated() {
        this.setGradient(this.weather, 'todayC');
        this.setGradient(this.weather, 'tomorrowC');
        this.setGradient(this.weather, 'afterC');
    },
    methods: {
        setGradient(weather, day) {
            console.log(day);
            let dayF = day;
            console.log(dayF);
            let element = document.querySelector(`.colors.${day}`);
            console.log(element);
            if (`weather.${day}`) {
                let temp = parseInt(`weather.${day}`);
                if (`weather.${day}` === "C" && temp < 15)
                    element.classList.add("background-blue");
                else if (`weather.${day}` === "C" && temp > 35)
                    element.classList.add("background-red");
                else element.classList.add("background-yellow");
            }
        }
    }
};
</script>

<style >
.icon-font {
    font-family: 'MeteoconsRegular';
}

.colors {
    background-color: #efebe7c2;
}

.background-yellow {
    background-color: #e3b612ad;
}

.tomorrowC.background-yellow {
    background-color: #facc05cf;
}

.afterC.background-yellow {
    background-color: #b79404e8;
}

</style>
