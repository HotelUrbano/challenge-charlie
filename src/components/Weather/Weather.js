import React from 'react';
import css from './Weather.module.css';

import Header from '../Header/Header';
import Weekday from '../Weekday/Weekday';

export default ({ weather, onSearch, onUnit }) => {
  console.log(weather && weather.tomorrow.temp);
  return (
    <div className={css.Weather}>
      <Header location="Rio de Janeiro" onSearch={onSearch} />
      <Weekday name="Hoje" temp={weather && weather.today.temp} color="#DFA800" onUnit={onUnit} />
      <Weekday name="Amanhã" temp={weather && weather.tomorrow.temp} color="#B9264F" onUnit={onUnit} />
      <Weekday name="Depois de Amanhã" temp={weather && weather.after.temp} color="#1F88A7" onUnit={onUnit} />
    </div>
  );
}
