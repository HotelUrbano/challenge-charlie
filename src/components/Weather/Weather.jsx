import React from 'react'
import PropTypes from 'prop-types'
import MainWeather from 'components/MainWeather/MainWeather'
import WeatherDetails from 'components/WeatherDetails/WeatherDetails'
import styles from './Weather.module.scss'

const Weather = ({ day, weather }) => (
  <div className={styles.weather}>
    <MainWeather day={day} temperature={weather.temperature} />

    {weather.description && (
      <WeatherDetails
        description={weather.description}
        humidity={weather.humidity}
        pressure={weather.pressure}
        windSpeed={weather.wind.speed}
      />
    )}
  </div>
)

Weather.propTypes = {
  day: PropTypes.string.isRequired,
  weather: PropTypes.object.isRequired,
}

export default Weather
