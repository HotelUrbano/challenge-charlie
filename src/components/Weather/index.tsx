import React, { useMemo, useEffect, useState, useCallback } from 'react';

import { Container } from './styles';
import Icon from '../Icon';

interface Weather {
  description: string;
  icon: string;
}

interface WeatherDay {
  humidity: number;
  pressure: number;
  speed: number;
  temp: {
    day: number;
  };
  weather: Weather[];
}

interface WeatherProps {
  weather: WeatherDay;
  active: boolean;
  title: string;
  degrade: number;
  onClick(): void;
}

const Weather: React.FC<WeatherProps> = ({
  title,
  weather,
  active,
  degrade,
  onClick,
}) => {
  const [temperature, setTemperature] = useState(100);
  const [temperatureType, setTemperatureType] = useState<'C' | 'F'>('C');

  useEffect((): void => {
    if (!weather.temp) {
      return;
    }
    setTemperature(Math.trunc(weather.temp.day));
  }, [weather.temp]);

  const weatherDescription = useMemo((): string => {
    const fetchedDescription = weather.weather;

    if (!fetchedDescription) {
      return '';
    }

    return fetchedDescription[0].description.toUpperCase();
  }, [weather]);

  const iconName = useMemo((): string => {
    const icon = weather.weather;

    if (!icon) {
      return '';
    }
    return icon[0].icon;
  }, [weather]);

  const windSpeedFormattedToKMPerHour = useMemo(() => {
    const windFormatted = weather.speed * 3.6;
    return windFormatted.toFixed(2);
  }, [weather.speed]);

  const temperatureInFahrenheit = useMemo(() => {
    const result = (temperature * 9) / 5 + 32;
    return Math.trunc(result);
  }, [temperature]);

  const toggleCelsiusToFahrenheit = useCallback(() => {
    temperatureType === 'C' ? setTemperatureType('F') : setTemperatureType('C');
  }, [temperatureType]);

  return (
    <Container
      active={active}
      onClick={onClick}
      temperature={temperature}
      degrade={degrade}
    >
      <Icon name={iconName} />

      <div>
        <h1>{title}</h1>
        {weather.temp && (
          <>
            <button type="button" onClick={toggleCelsiusToFahrenheit}>
              {temperatureType === 'C'
                ? `${temperature}ºC`
                : `${temperatureInFahrenheit}ºF`}
            </button>

            <fieldset>
              <h1>{weatherDescription}</h1>
              <p>Vento: {windSpeedFormattedToKMPerHour}km/h</p>
              <p>Humidade: {weather.humidity}</p>
              <p>Pressão: {weather.pressure}</p>
            </fieldset>
          </>
        )}
      </div>
    </Container>
  );
};

export default Weather;
