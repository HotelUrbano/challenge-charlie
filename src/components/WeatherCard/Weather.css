.icon:before {
    font-family: 'MeteoconsRegular';
    content: attr(data-icon);
    font-style: normal;
}

.weather_card {
    display: flex;
    align-items: center;
    justify-content: space-between;

    padding: 1.5rem 2rem;
    color: var(--white);
}

.weather_card .icon {
    font-size: 16rem;
} 

.weather_card .box_weather_data {
    display: flex;
    flex: 1;
    flex-direction: column;
    align-self: stretch;
    justify-content: space-around;
    margin-left: 4rem;
}

.weather_card .box_weather_data h2,
.weather_card .box_weather_data strong {
    font-size: 2rem;
    font-weight: 700;
    text-transform: uppercase;
} 

.weather_card .box_weather_data h3 {
    font-size: 1.8rem;
    font-weight: 500;
    text-transform: capitalize;
}

.weather_card .box_weather_data strong {
    cursor: pointer;
}

.weather_card .box_weather_data .box_more_info {
    display: block;
}

.weather_card .box_weather_data .box_more_info .weather_list_extra_data { 
    list-style: none;
    margin-top: 1.5rem;
}

.weather_card .box_weather_data .box_more_info .weather_list_extra_data li {
    font-size: 1.4rem;
    line-height: 2rem;
}

.weather_card.tomorrow .box_more_info,
.weather_card.after_tomorrow .box_more_info {
    display: none;
}

.weather_card.tomorrow .icon,
.weather_card.after_tomorrow .icon {
    font-size: 6rem;
    width: 24.5%;
    text-align: center;
}

.weather_card.today.yellow {
    background: var(--yellow-100);
}

.weather_card.today.blue {
    background: var(--blue-100);
}

.weather_card.today.red {
    background: var(--red-100);
}

.weather_card.tomorrow.yellow {
    background: var(--yellow-300);
}

.weather_card.tomorrow.blue {
    background: var(--blue-300);
}

.weather_card.tomorrow.red {
    background: var(--red-300);
}

.weather_card.after_tomorrow.yellow {
    background: var(--yellow-500);
}

.weather_card.after_tomorrow.blue {
    background: var(--blue-500);
}

.weather_card.after_tomorrow.red {
    background: var(--red-500);
}