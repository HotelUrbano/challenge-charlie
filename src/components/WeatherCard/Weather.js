import React from 'react';
import weatherIcons from '../../helpers/weatherIcons';
import windDirection from '../../helpers/windDirection';
import windSpeedConverter from '../../helpers/windSpeedConverter';
import temperatureColorChange from '../../helpers/temperatureColorChange';
import $ from 'jquery';

import './Weather.css';

export default function Weather(props) {

    function changeTemperature(temp, e) {
        if ($(e.target).attr("data-temp") === "f") {
            let arr = $(e.target).text().split("F");
            let fa = (arr[0] - 32) * 5 / 9;
            fa = fa.toFixed(0);
            $(e.target).text(fa + "°C");
            $(e.target).attr("data-temp", "c");
        } else {
            let fa = (temp * 9 / 5) + 32;
            fa = fa.toFixed(0);
            $(e.target).text(fa + " F");
            $(e.target).attr("data-temp", "f");
        }
    }

    return (
        <div className={`weather_card ${props.className} ${temperatureColorChange(props.temperature)}`}>
            <i className="icon" data-icon={weatherIcons(props.icon)}></i>
            <div className="box_weather_data">
                <h2>{props.day}</h2>
                <strong onClick={e => changeTemperature(props.temperature, e)} data-temp="c">{parseFloat(props.temperature).toFixed(0)}°C</strong>
                <div className="box_more_info">
                    <h3>{props.description}</h3>
                    <ul className="weather_list_extra_data">
                        <li>Vento: {windDirection(props.windDirection)} {windSpeedConverter(props.windVelocity).toFixed(0)}km/h</li>
                        <li>Humidade: {props.humidity}%</li>
                        <li>Pressão: {props.pressure}hPA</li>
                    </ul>
                </div>
            </div>
        </div>
    );
}