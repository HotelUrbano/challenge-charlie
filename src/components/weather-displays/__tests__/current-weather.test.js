import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Weather from "../current-weather";

describe("weather display", () => {
    const props = {
        temp_max: 30,
        humidity: 78,
        pressure: 1003,
        speed: 12,
        deg: 80,
        main: "Clear",
        unit: "C",
        toggleUnit: () => {}
    };

    it("matches snapshot", () => {
        const component = renderer.create(<Weather {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("shows correct information", () => {
        const wrapper = mount(<Weather {...props} />);

        expect(wrapper.find("span.temperature").text()).toBe("30 ºC");
        expect(wrapper.find("span.wind").text()).toBe("Vento: Leste 12 Kmph");
        expect(wrapper.find("span.humidity").text()).toBe("Humidade: 78 %");
        expect(wrapper.find("span.pressure").text()).toBe("Pressão: 1003 hPA");
    });
});
