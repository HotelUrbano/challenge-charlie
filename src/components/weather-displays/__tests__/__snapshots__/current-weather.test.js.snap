// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather display matches snapshot 1`] = `
<div
  className="sc-bdVaJa gAhrVP"
>
  <div>
    <img
      alt="ícone para a condição climática"
      className="sc-bwzfXH fzlwxk"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "clear.svg",
        }
      }
    />
  </div>
  <div>
    <h2>
      Hoje
    </h2>
    <p>
      <span
        className="temperature"
        onClick={[Function]}
      >
        30
         º
        C
      </span>
    </p>
    <h2>
      Ensolarado
    </h2>
    <p>
      <span
        className="wind"
      >
        Vento: 
        Leste
         
        12
         Kmph
      </span>
      <br />
      <span
        className="humidity"
      >
        Humidade: 
        78
         %
      </span>
      <br />
      <span
        className="pressure"
      >
        Pressão: 
        1003
         hPA
      </span>
    </p>
  </div>
</div>
`;
